src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -0,0 +1,196 @@
+                        )
+                    )
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to register local model");
+                registerLocalModelFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+            }
+        };
+
+        String modelName = null;
+        String modelVersion = null;
+        String description = null;
+        MLModelFormat modelFormat = null;
+        String modelGroupId = null;
+        String modelContentHashValue = null;
+        String modelType = null;
+        String embeddingDimension = null;
+        FrameworkType frameworkType = null;
+        String allConfig = null;
+        String url = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelName = (String) content.get(NAME_FIELD);
+                        break;
+                    case VERSION_FIELD:
+                        modelVersion = (String) content.get(VERSION_FIELD);
+                        break;
+                    case DESCRIPTION_FIELD:
+                        description = (String) content.get(DESCRIPTION_FIELD);
+                        break;
+                    case MODEL_FORMAT:
+                        modelFormat = MLModelFormat.from((String) content.get(MODEL_FORMAT));
+                        break;
+                    case MODEL_GROUP_ID:
+                        modelGroupId = (String) content.get(MODEL_GROUP_ID);
+                        break;
+                    case MODEL_TYPE:
+                        modelType = (String) content.get(MODEL_TYPE);
+                        break;
+                    case EMBEDDING_DIMENSION:
+                        embeddingDimension = (String) content.get(EMBEDDING_DIMENSION);
+                        break;
+                    case FRAMEWORK_TYPE:
+                        frameworkType = FrameworkType.from((String) content.get(FRAMEWORK_TYPE));
+                        break;
+                    case ALL_CONFIG:
+                        allConfig = (String) content.get(ALL_CONFIG);
+                        break;
+                    case MODEL_CONTENT_HASH_VALUE:
+                        modelContentHashValue = (String) content.get(MODEL_CONTENT_HASH_VALUE);
+                        break;
+                    case URL:
+                        url = (String) content.get(URL);
+                        break;
+                    default:
+                        break;
+
+                }
+            }
+        }
+
+        if (Stream.of(modelName, modelVersion, modelFormat, modelGroupId, embeddingDimension, frameworkType, modelContentHashValue, url)
```

Comment:
`embeddingDimension` is not a mandatory field based on [this](https://opensearch.org/docs/latest/ml-commons-plugin/api/#request-fields). Also, since mlConfig is not mandatory, we can't have `embeddingDimension` mandatory.

Reply:
That documentation clearly states that the only optional field in model config is `all_config`, the remaining fields are required

Reply:
Ah! You are right. I missed the descriptionof it.

Reply:
Even the modelType is a mandatory field, right?

Reply:
ah missed that too

Reply:
Checking these params requirement can be a little confusing.